### PR TITLE
Dev -> Master for v2.4 release

### PR DESCRIPTION
**Pull request by ewels**
## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/README.md [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
-
-[![Nextflow](https://img.shields.io/badge/nextflow%20DSL2-%E2%89%A521.10.3-23aa62.svg?labelColor=000000)](https://www.nextflow.io/)
-[![run with conda](http://img.shields.io/badge/run%20with-conda-3EB049?labelColor=000000&logo=anaconda)](https://docs.conda.io/en/latest/)
-[![run with docker](https://img.shields.io/badge/run%20with-docker-0db7ed?labelColor=000000&logo=docker)](https://www.docker.com/)
-[![run with singularity](https://img.shields.io/badge/run%20with-singularity-1d355c.svg?labelColor=000000)](https://sylabs.io/docs/)
-
-[![Get help on Slack](http://img.shields.io/badge/slack-nf--core%20%23{{ short_name }}-4A154B?labelColor=000000&logo=slack)](https://nfcore.slack.com/channels/{{ short_name }})
-[![Follow on Twitter](http://img.shields.io/badge/twitter-%40nf__core-1DA1F2?labelColor=000000&logo=twitter)](https://twitter.com/nf_core)
-[![Watch on YouTube](http://img.shields.io/badge/youtube-nf--core-FF0000?labelColor=000000&logo=youtube)](https://www.youtube.com/c/nf-core)
+[![AWS CI](https://img.shields.io/badge/CI%20tests-full%20size-FF9900?logo=Amazon%20AWS)](https://nf-co.re/{{ short_name }}/results)
```
Comment (FriederikeHanssen) [inline nf_core/pipeline-template/README.md:5]: Sorry, I know I approved the previous PR 🙈 . But the badges for me are not displayed properly 
Screenshot 2022-05-13 at 17 22 22

Comment (ewels) [reply nf_core/pipeline-template/README.md:5]: Yeah they won't with the raw template. You need to test with `nf-core create`

Even then some of the badges (GitHub) won't display properly unless you call your test pipeline the same name as an existing pipeline repo. eg. `rnaseq`

Comment (FriederikeHanssen) [reply nf_core/pipeline-template/README.md:5]: ah my bad, didn't realise I was in the template folder already. I should go home 😴

Comment (ewels) [reply nf_core/pipeline-template/README.md:5]: This is why we don't release on Friday evenings 😅

**Review by FriederikeHanssen — COMMENTED** [review]
Don't think I know enough python to approve. I tried syncing the testpipeline though, and that worked really well: https://github.com/nf-core/testpipeline/pull/40
(Only fail, because I adapted the linting.yml to use the dev version. then it doesn't match the template and the linting rightfully complains)

**Comment by ewels** [comment]
Oh nice, thanks! I want to do a sync test where it automatically creates the PR for me still (it's that stuff with the GitHub API that I refactored) but a relief to know that the sync itself isn't playing up 😅

**Review by drpatelh — APPROVED** [review]
🚀 What could possibly go wrong! 😆

**Review by mirpedrol — APPROVED** [review]
🚀

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +50,7 @@ Very minor patch release to fix the full size AWS tests and re-run the template
 - Remove traces of markdownlint in the template ([#1486](https://github.com/nf-core/tools/pull/1486)
 - Remove accidentally added line in `CHANGELOG.md` in the template ([#1487](https://github.com/nf-core/tools/pull/1487))
 - Update linting to check that `.editorconfig` is there and `.yamllint.yml` isn't.
+- Don't save md5sum for `versions.yml` when running `nf-core modules create-test-yml` ([#1511](https://github.com/nf-core/tools/pull/1511))
```
Comment (JoseEspinosa) [inline CHANGELOG.md:53]: Is correct that this change is under the previous tools version?

**Review by JoseEspinosa — APPROVED** [review]
Awesome job! 
Though probably an opinion of a real pythonist would be more relevant :stuck_out_tongue:

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ewels) [reply CHANGELOG.md:53]: Yup, well spotted both - fixed https://github.com/nf-core/tools/pull/1583

Comment (ewels) [reply CHANGELOG.md:53]: Yup, well spotted both - fixed https://github.com/nf-core/tools/pull/1583

File: .github/workflows/fix-linting.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: Fix linting from a comment
```
[the remark this answers is not in the document]
Comment (ewels) [reply .github/workflows/fix-linting.yml:1]: Yup!

[the remark this answers is not in the document]
Comment (ewels) [reply .github/workflows/fix-linting.yml:1]: Been using it on the website repo for a few weeks now and it's as amazing as it sounds.. 🤩

**Pull request merged**